templates/lib/herb/ast/nodes.rb.erb

```diff
@@ -1,9 +1,17 @@
 module Herb
   module AST
     <%- nodes.each do |node| -%>
+    #: type serialized_<%= node.human %> = {
+      <%- node.fields.each do |field| -%>
+      <%- is_nilable = !%w[Array[Herb::AST::Node] Array[Herb::AST::ERBWhenNode] Array[Herb::AST::ERBInNode] bool nil].include?(field.ruby_type) -%>
```

Comment:
As we talked about in Euruko, everything that isn't an array, or a boolean, or nil itself is nilable. @marcoroth